src/dump.c

```diff
@@ -205,14 +205,26 @@ static void jl_load_sysimg_so(char *fname)
 #else
         if (dladdr((void*)sysimg_gvars, &dlinfo) != 0) {
             jl_sysimage_base = (intptr_t)dlinfo.dli_fbase;
-        } else {
+        }
+        else {
             jl_sysimage_base = 0;
         }
 #endif
+        const char *sysimg_data = (const char*)jl_dlsym(jl_sysimg_handle, "jl_system_image_data");
+        size_t len = *(size_t*)jl_dlsym(jl_sysimg_handle, "jl_system_image_size");
+        jl_restore_system_image(sysimg_data, len);
     }
     else {
-        sysimg_gvars = 0;
+        jl_error("Could not load system image.");
     }
+    int build_mode = 0;
+#ifdef _OS_WINDOWS_
+    //XXX: the windows linker forces our system image to be
+    //     linked against only one dll, I picked libjulia-release
+    if (jl_is_debugbuild()) build_mode = 1;
```

Comment:
i don't think this test is meaningful here. we've already loaded sys.dll, so we're already in a bit of trouble if we happen to be julia-debug.exe on windows.